index.d.ts

```diff
@@ -0,0 +1,25 @@
+declare module 'wheel-indicator' {
+    type WheelIndicatorEvent = Event & {
+        direction: string;
+    }
+
+    interface Options {
+        elem?: HTMLElement | string;
+        callback?: (event: WheelIndicatorEvent) => void;
+        preventMouse?: boolean;
+    }
+
+    export default class WheelIndicator {
+        constructor(options: Options);
+
+        destroy(): WheelIndicator;
+
+        getOption(option: keyof Options): any;
```

Comment:
I don't know all features of typescript well, but i believe here should be some generic, that declares, that returnable type depends on option. I found this solution, but not sure that it's right:
type ValueOf<T> = T[keyof T];
...
getOption(option: keyof Options): ValueOf<Options>;

Reply:
look like work decision
https://www.typescriptlang.org/play/index.html?ssl=16&ssc=1&pln=45&pc=1#code/C4TwDgpgBAaghgGwK4QPIDMA8AVAfFAXimwG0BrCEAe3WIF0BuAKCYEsA7YCAJ3TgGNoASXYBnYHHaCoAbyhMoiqOzgBbCAC4o47hwDmzJVDh7NypKoBGPZgF8W-KmOBQO4yYK0j3U6ERkKSirqWgDkcKEANIGKJmYAjABMTPZM6EhSwKxOUKbAqGBZTgAUVIXZ7FoU1LTeEr4AlFrwyGhYdR4Q+HIxUNwQwEjc7K7OnSRlRex0KUxAA